gems/pending/VMwareWebService/VimService.rb

```diff
@@ -903,20 +903,33 @@ def retrievePropertiesEx(propCol, specSet, max_objects = nil)
     (parse_response(response, 'RetrievePropertiesExResponse')['returnval'])
   end
 
-  def retrievePropertiesCompat(propCol, specSet, max_objects = nil)
-    oc = VimArray.new('ArrayOfObjectContent')
-
-    rv = retrievePropertiesEx(propCol, specSet, max_objects)
-    if rv
-      oc.concat(rv['objects'])
+  def retrievePropertiesIter(propCol, specSet, max_objects = nil)
+    result = retrievePropertiesEx(propCol, specSet, max_objects)
```

Comment:
If you add this here
```ruby
return to_enum(:retrievePropertiesIter, propCol, specSet, max_objects) unless block_given?
```

you can use all the `Enumerator` and `Enumerable` methods on your method too, which could make the `retrievePropertiesCompat` below a one liner :smile: 

but that's probably just sugar, as you are using this method only in one place?!